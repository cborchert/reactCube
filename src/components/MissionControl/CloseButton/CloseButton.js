import React from "react";
import "./CloseButton.scss";

const CloseButton = ({ close = () => {} }) => (
  <button className="CloseButton" onClick={close} data-testid="close-button">
    âœ–
  </button>
);
export default CloseButton;
