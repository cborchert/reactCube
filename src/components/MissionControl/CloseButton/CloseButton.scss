.CloseButton {
  position: fixed;
  top: 10px;
  right: 10px;
  opacity: 0.6;
  font-size: 24px;
  color: black;
  background-color: transparent;
  outline: none;
  border: none;
  opacity: 1;
  transition: 200ms;
  z-index: 4;
  &:hover {
    transform: rotateZ(90deg);
    cursor: pointer;
  }
}
