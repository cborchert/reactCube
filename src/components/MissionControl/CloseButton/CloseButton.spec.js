import React from "react";
import { render, fireEvent, cleanup } from "@testing-library/react";
import CloseButton from "./CloseButton";

afterEach(cleanup);

describe("CloseButton", () => {
  it("renders correctly", () => {
    const { container } = render(<CloseButton />);
    expect(container).toMatchSnapshot();
  });
  it("fires click handler on click", () => {
    const myClickHandler = jest.fn();
    const { getByTestId } = render(<CloseButton close={myClickHandler} />);
    fireEvent.click(getByTestId("close-button"));
    expect(myClickHandler).toHaveBeenCalledTimes(1);
  });
});
