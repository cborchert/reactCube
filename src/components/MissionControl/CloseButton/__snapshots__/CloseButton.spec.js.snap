// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CloseButton renders correctly 1`] = `
<div>
  <button
    class="CloseButton"
    data-testid="close-button"
  >
    âœ–
  </button>
</div>
`;
